selfservice/strategy/oidc/provider_patreon.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/herodot"
+)
+
+type ProviderPatreon struct {
+	config *Configuration
+	reg    dependencies
+}
+
+type PatreonIdentityResponse struct {
+	Data struct {
+		Attributes struct {
+			Email     string `json:"email"`
+			FirstName string `json:"first_name"`
+			FullName  string `json:"full_name"`
+			ImageUrl  string `json:"image_url"`
+			LastName  string `json:"last_name"`
+		} `json:"attributes"`
+		Id   string `json:"id"`
+		Type string `json:"type"`
+	} `json:"data"`
+}
+
+func NewProviderPatreon(
+	config *Configuration,
+	reg dependencies,
+) *ProviderPatreon {
+	return &ProviderPatreon{
+		config: config,
+		reg:    reg,
+	}
+}
+
+func (d *ProviderPatreon) Config() *Configuration {
+	return d.config
+}
+
+func (d *ProviderPatreon) oauth2(ctx context.Context) *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     d.config.ClientID,
+		ClientSecret: d.config.ClientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  "https://www.patreon.com/oauth2/authorize",
+			TokenURL: "https://www.patreon.com/api/oauth2/token",
+		},
+		RedirectURL: d.config.Redir(d.reg.Config().OIDCRedirectURIBase(ctx)),
+		Scopes:      d.config.Scope,
+	}
+}
+
+func (d *ProviderPatreon) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return d.oauth2(ctx), nil
+}
+
+func (d *ProviderPatreon) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	if isForced(r) {
+		return []oauth2.AuthCodeOption{
+			oauth2.SetAuthURLParam("prompt", "consent"),
+		}
+	}
+	return []oauth2.AuthCodeOption{
+		oauth2.SetAuthURLParam("prompt", "none"),
+	}
+}
+
+func (d *ProviderPatreon) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+	identityUrl := "https://www.patreon.com/api/oauth2/v2/identity?fields%5Buser%5D=first_name,last_name,url,full_name,email,image_url"
+
+	cl := http.Client{}
+	req, _ := http.NewRequest("GET", identityUrl, nil)
```

Comment:
Please use the resilient client, similar to https://github.com/ory/kratos/blob/f2960124cf62886025b775a06c85e697270f8d6a/selfservice/strategy/oidc/provider_auth0.go#L79
It uses some retry logic. Also, you should handle the error here (although it should never error) to prevent a later update to break this.

Reply:
I made the requested change :)

Do the maintainers want me to resolve conversations when I make the change or do the maintainers want me to leave them open for you to change when you have reviewed the change?

Reply:
Typically, the creator of the comment will resolve it themselves, after whether the comment is resolved. :)